core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -649,6 +654,9 @@ boolean isPreserveAliases() {
                     wildcardSeen = true;
                 }
             }
+            if(isDeprecated) {
+              DEPRECATION_LOGGER.deprecated("use of + is deprecated in index names as it is implicit");
```

Comment:
change the message to "Support for '+' as part of index expressions is deprecated" ?